batch/processor_Notifications_Token.py

```diff
@@ -423,37 +433,42 @@ def db_merge(
             notification.notification_type = self.notification_type
             notification.priority = 0
             notification.address = entry["args"]["from"]
-            notification.block_timestamp = self._gen_block_timestamp(entry)
+            notification.block_timestamp = await self._gen_block_timestamp(entry)
             notification.args = dict(entry["args"])
             notification.metainfo = metadata
-            db_session.merge(notification)
+            await db_session.merge(notification)
 
 
 # メイン処理
-def main():
+async def main():
     watchers = [
         WatchTransfer(),
         WatchApplyForTransfer(),
         WatchApproveTransfer(),
         WatchCancelTransfer(),
     ]
 
-    e = ThreadPoolExecutor(max_workers=WORKER_COUNT)
     LOG.info("Service started successfully")
 
     while True:
         start_time = time.time()
 
-        fs = []
-        for watcher in watchers:
-            fs.append(e.submit(watcher.loop))
-        wait_all_futures(fs)
+        try:
+            tasks = await SemaphoreTaskGroup.run(
+                *[watcher.loop() for watcher in watchers], max_concurrency=WORKER_COUNT
+            )
+            [task.result() for task in tasks]
+        except ExceptionGroup as e:
+            LOG.error(e)
```

Comment:
The multi-process runner has been replaced with an async runner that uses a single process.